internal/app/upstream/client.go

```diff
@@ -289,3 +291,19 @@ func shutDown(ctx context.Context, conn *grpc.ClientConn) {
 func (e *UnsupportedResourceError) Error() string {
 	return fmt.Sprintf("Unsupported resource typeUrl: %s", e.TypeURL)
 }
+
+func updateConnectivityMetric(ctx context.Context, conn *grpc.ClientConn, scope tally.Scope) {
+	connectedStat := "connected"
+	for {
```

Comment:
Should we put a sleep here to poll every 1s instead? or is this blocking on `WaitForStateChange`?

Reply:
its a blocked channel, we woudn;t need to poll.